packages/environment-check/src/formatter.ts

```diff
@@ -23,3 +23,36 @@ export function countNumberOfServices(catalogResult?: ODataServiceInfo[]): numbe
 export function getServiceCountText(count: number) {
     return count === 1 ? `${count} service` : `${count} services`;
 }
+
+/**
+ * Returns replacer function that can be used with JSON.stringify to detect
+ * and replace circular structures.
+ *
+ * @example JSON.stringify(object, getCircularReplacer());
+ * @returns - replacer that replaces circular structures
+ */
+export function getCircularReplacer(): (key: string, value: any) => any {
+    const stack: any[] = [];
+    const keys: string[] = [];
+    return function replacer(key, value) {
+        if (stack.length > 0) {
+            const thisPos = stack.indexOf(this);
```

Comment:
Use `pos` as a name instead of `thisPos`?

Reply:
changed it to the simplified algorithm